util/logging/retrieve-cluster-logs.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import gzip
+import json
+import logging
+import os
+import sys
+import tarfile
+import tempfile
+import time
+from datetime import datetime
+
+import argparse
+import boto3
+
+from pcluster.cloudwatch_logs.utils import get_log_group_name
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__file__)
+
+
+def err_and_exit(message):
+    """Log the given error message and exit nonzero."""
+    LOGGER.error(message)
+    sys.exit(1)
+
+
+def get_default_bucket_prefix(cluster_name):
+    """Get default keypath to store logs under in an s3 bucket for the given cluster."""
+    return "{cluster_name}-logs-{end_time}".format(cluster_name=cluster_name, end_time=datetime.now().isoformat())
```

Comment:
what about this datetime? Shall we have it always in utc and not in local time?

Reply:
How about I just make it a timestamp as well? The main point of including it there was just to make sure that the output file. `end_time` is a bit misleading in that way.

Reply:
That's done.